geth/jail/jail_cell.go

```diff
@@ -43,29 +36,26 @@ func newJailCell(id string, vm *otto.Otto, lo *loop.Loop) (*JailCell, error) {
 		return nil, err
 	}
 
-	return &JailCell{
-		id:  id,
-		vm:  vm,
-		lo:  lo,
-		sem: semaphore.New(1, JailCellRequestTimeout*time.Second),
+	return &JCell{
+		id: id,
+		vm: vm,
+		lo: lo,
 	}, nil
 }
 
-// Copy returns a new JailCell instance with a new eventloop runtime associated with
-// the given cell.
-func (cell *JailCell) Copy() (common.JailCell, error) {
-	vmCopy := cell.vm.Copy()
-	return newJailCell(cell.id, vmCopy, loop.New(vmCopy))
-}
-
 // Fetch attempts to call the underline Fetch API added through the
 // ottoext package.
-func (cell *JailCell) Fetch(url string, callback func(otto.Value)) (otto.Value, error) {
+func (cell *JCell) Fetch(url string, callback func(otto.Value)) (otto.Value, error) {
+	cell.Lock()
```

Comment:
I suggest we add `defer` and call `cell.vm.Run` at the end as this combination looks more straightforward for understanding and less error-prone.